packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -547,13 +550,37 @@ export default class CosmeticFilterBucket {
         // Dispatch rules in `injections` or `styles` depending on type
         if (rule.isScriptInject() === true) {
           if (getInjectionRules === true && injectionsDisabled === false) {
+            if (emitOnFiltersEngine !== undefined) {
+              emitOnFiltersEngine('script-rule-matched', rule, {
+                domain,
+                classes,
+                hrefs,
+                ids,
+              });
+            }
             injections.push(rule);
           }
         } else if (rule.isExtended()) {
           if (getExtendedRules === true) {
+            if (emitOnFiltersEngine !== undefined) {
+              emitOnFiltersEngine('extended-rule-matched', rule, {
+                domain,
+                classes,
+                hrefs,
+                ids,
+              });
+            }
             extended.push(rule);
           }
         } else {
+          if (emitOnFiltersEngine !== undefined) {
+            emitOnFiltersEngine('style-rule-matched', rule, {
+              domain,
```

Comment:
The CosmeticFilterContext need some more precise identifier so we can distinguish which exact tab was targeted. 

I'm afraid there is no other way than to pass some identifier (like tabId, but variable name should be generic) as argument to `getCosmeticsFilters` so it can be passed to the emited event.

Reply:
I guess we could add similar identifier events emited for network filters